### PR TITLE
Fix up a couple of SEO issues

### PR DESCRIPTION
**Pull request by jrdh**
Add og:url and canonical urls to specimen pages.
Add jsonld to specimen pages, though not quite in line with [the recommendations](https://developers.google.com/search/docs/guides/intro-structured-data?hl=en#structured-data) as I wasn't sure how to map to the [schema.org](https://schema.org/docs/full.html) context options. That requires a bit more thought me thinks!

### CODE REVIEW

**Comment by rdmpage** [comment]
Hi @jrdh many thanks for added these tags! They make my life simpler (and may help with SEO). I'm working on a tool to display links to entities shown on a web page (e.g., is there a paper that cites this specimen?) and knowing what entity is the subject of the web page is a crucial step.

I also see that you're now exposing JSON-LD - cool! It seems pretty clear that this is the way forward.

Couple of comments. The field `dwc:associatedMedia` often has pipe-delimited URLs (e.g., `"https://www.nhm.ac.uk/services/media-store/asset/cafa7a935cb5c705a6999f60ff66f82cc034573e/contents/preview | https://www.nhm.ac.uk/services/media-store/asset/57d3a4275b00b46f8dc9993331b18ed732bf25aa/contents/preview"`), which are just yuck. Is there any way when you generate RDF that you could parse these into separate fields so that 'dwc:associatedMedia` becomes an array of URLs?

Oh, and I'm pretty sure that there is no such thing as `"dc:type": "Specimen", I guess this is just a text term, but would be better if it was an actual Darwin Core term. Ironically, there isn't a nice generic term for specimen (sigh).

Lastly, re the JSON-LD, any thoughts on "framing"? Stop me at any point if I'm telling you stuff you already know (I'm still getting my head around JSON-LD's many nuances), but there's a way to output the JSON-LD so that everything is part of the same document tree (rooted, in this case, on the specimen). So the two images would be included underneath `foaf:depiction` (which at the moment just has the `@id` member). This means anyone processing the JSON-LD can treat it as regular JSON, not have to go looking in the graph for the images. Hope this makes sense.

**Comment by jrdh** [comment]
Hi @rdmpage, not a problem at all, thanks for pointing out things we can improve :slightly_smiling_face:, looking forward to seeing your linker, it sounds interesting.

Those pipe-separated lists are indeed yuck. I'll change that.

You're right, there's no such thing as `Specimen` in the `dc:type` vocab. This is an interesting one and I think I've misinterpreted part of the [CSPP](https://cetafidentifiers.biowikifarm.net/wiki/CETAF_Specimen_Preview_Profile_(CSPP)) _and_ mis-mapped a field (specifically the `kindOfMaterial` CSPP element) as this seems like it should be referring to the "nature of the specimen", though none of the examples given align with the `dc:type` vocabulary (`herbarium sheet`, `fossil`, `conserved in alcohol`). Perhaps this should be [PhysicalObject](https://www.dublincore.org/specifications/dublin-core/dcmi-terms/dcmitype/PhysicalObject/)? Or perhaps we should, as you say, use something from dwc, maybe `basisOfRecord` and/or `preparations`. We'd need to know what the intent of that `kindOfMaterial` property in the CSPP was to know what would be more appropriate probably.

And I'll have a look at the framing as I agree it's hard to parse and given that it's primary focus is being machine-readable this isn't ideal... I think I looked at peer CETAF institutions and matched their outputs but I'll double-check and see what I can do about ours.

Thanks again for raising the original issues and these finer tuned points, much appreciated!
